tests/test_attentions.py

```diff
@@ -81,7 +96,9 @@ def test_order_invariance(
     )
 
     # Check that a shuffled input produces the same results
-    seqs = [SEQ, SEQ - 16] if attention_name != "blocksparse" else [SEQ]
+    seqs = (
+        [SEQ, SEQ - 16] if (attention_name != "blocksparse" and not causal) else [SEQ]
```

Comment:
ah, good question ! Basically we adjust the mask on the fly when the sequence is smaller than initially thought, but we don't have the codepath in place to also adjust the causal mask so that broke (it "worked" before because we were not actually applying the causal mask..). I can fix that in this PR and then refactor all this part in a sequel ?

Reply:
Ah okay thanks for the explanation :) That sounds good

Reply:
Ah okay thanks for the explanation :) That sounds good